java/code/src/com/redhat/rhn/domain/contentmgmt/FilterCriteria.java

```diff
@@ -89,13 +100,34 @@ public FilterCriteria() {
      * @param matcherIn the matcher type
      * @param fieldIn the field to match
      * @param valueIn the match value
+     * @throws IllegalArgumentException when the matcher-field combination is not allowed
      */
     public FilterCriteria(Matcher matcherIn, String fieldIn, String valueIn) {
+        validate(matcherIn, fieldIn);
         this.matcher = matcherIn;
         this.field = fieldIn;
         this.value = valueIn;
     }
 
+    /**
+     * Validate the matcher-field combination
+     *
+     * @param matcher the matcher
+     * @param field the field
+     * @throws IllegalArgumentException when validation does not pass
+     */
+    public static void validate(Matcher matcher, String field) throws IllegalArgumentException {
+        if (!validCombinations.contains(Pair.of(matcher, field))) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid criteria combination (matcher: '%s', field: '%s')", matcher, field));
+        }
+    }
+
+    // todo remove
```

Comment:
I think this is intended to be removed before merging then :wink:

Reply:
ufff, sure thing!!! I totally forgot, thanks for this catch :sweat_smile: